testng-core-api/src/main/java/org/testng/ITestNGMethod.java

```diff
@@ -220,6 +220,17 @@ default boolean hasAfterGroupsConfiguration() {
 
   void setInterceptedPriority(int priority);
 
+  /**
+   * Gets priority for class hierarchy sorting. There is no guaranty that methods will be sorted in
+   * a class hierarchy ordering, if dependsOnMethods or Groups are used they will have higher
+   * priority in the sort.
+   *
+   * @return class hierarchy priority
+   */
+  int getClassHierarchyPriority();
```

Comment:
I think we should perhaps extract out these two methods into an *internal only* interface and have the relevant `ITestNGMethod` implementations implement these two methods explicitly. We can add an explicit predicate within our comparator that applies this conditionally. Not sure if its a good idea to expose this when its purpose is to facilitate an internal functionality.

Reply:
Agree, it will be really nice. Same for `InterceptedPriority`, right?

Reply:
I have now implemented that in the original PR.